notifiers/providers/victorops.py

```diff
@@ -0,0 +1,79 @@
+from ..core import Provider
+from ..core import Response
+from ..utils import requests
+
+
+class VictorOps(Provider):
+    """Send VictorOps webhook notifications"""
+
+    base_url = "https://portal.victorops.com/ui/{ORGANIZATION_ID}/incidents"
+    site_url = "https://portal.victorops.com/dash/{ORGANIZATION_ID}#/advanced/rest"
+    name = "victorops"
+
+    _required = {
+        "required": [
+            "rest_url",
+            "message_type",
+            "entity_id",
+            "entity_display_name",
+            "state_message",
+        ]
+    }
+    _schema = {
+        "type": "object",
+        "properties": {
+            "rest_url": {
+                "type": "string",
+                "format": "uri",
+                "title": "the REST URL to use with routing_key. create one in victorops `integrations` tab.",
+            },
+            "message_type": {
+                "type": "string",
+                "title": "severity level can be: "
+                "- CRITICAL or WARNING: Triggers an incident "
+                "- ACKNOWLEDGEMENT: sends Acknowledgment to an incident "
+                "- INFO: Creates a timeline event but doesn't trigger an incident "
+                "- RECOVERY: Resolves an incident",
+            },
+            "entity_id": {
+                "type": "string",
+                "title": "Unique id for the incident for aggregation ,Acknowledging, or resolving.",
+            },
+            "entity_display_name": {
+                "type": "string",
+                "title": "Display Name in the UI and Notifications.",
+            },
+            "state_message": {
+                "type": "string",
+                "title": "This is the description that will be posted in the incident.",
+            },
+            "annotations": {
+                "type": "object",
+                "format": "{'annotation_type': 'annotation'}",
```

Comment:
format is just a name a gave the variable, its related to the way you pass annotations to victorops
there are 3 types of annotations: 

- vo_annotate.u.Runbook
- vo_annotate.s.Note
- vo_annotate.i.image

(`Runbook`, `Image` and `Note` are random names you can give your annotations)

Reply:
That's not accurate AFAIK, read about `format` here: https://json-schema.org/understanding-json-schema/reference/string.html#format

Also, there are other provider that use either JSONSchema out of the box formats or custom one that are implemented as part of notifiers.

If you want to limit the property to be of a closed list (but an enum is too basic) look into `oneOf`

If you'd share some examples of the expected format it would be very helpful.

edit: just saw the example from the test you added...

I think a more fitting schema would be this:

```json
{
	"annotations": {
		"type": "object",
		"patternProperties": {
			"vo_annotate.u.": {"type": "string"},
			"vo_annotate.s.": {"type": "string"},
			"vo_annotate.i.": {"type": "string"},
		},
		"minProperties": 1
}
```
WDYT?

Reply:
Your right, miss understood the concept of schema-json.
thanks (+ you gave me something to read about witch is fun :))